src/org/exist/storage/BrokerPool.java

```diff
@@ -871,6 +870,16 @@ protected void initialize() throws EXistException, DatabaseConfigurationExceptio
                         symbols = new SymbolTable(conf);
                         isReadOnly = isReadOnly || !symbols.getFile().canWrite();
 
+                        try {
+                            // initialize EXPath repository so indexManager and
+                            // startup triggers can access it
+                            expathRepo = ExistRepository.getRepository(this.conf);
+                        } catch(final PackageException e) {
+                            LOG.warn("Failed to initialize expath repository: " + e.getMessage() + " - " +
```

Comment:
I would just log an error but use an Optional as the reference type
instead, and then make future access to the Optional conditional to avoid
further NPEs

On 9 July 2015 at 12:51, Dmitriy Shabanov <notifications@github.com> wrote:

> In src/org/exist/storage/BrokerPool.java
> <https://github.com/eXist-db/exist/pull/726#discussion_r34247002>:
>
> > @@ -871,6 +870,16 @@ protected void initialize() throws EXistException, DatabaseConfigurationExceptio
> >                          symbols = new SymbolTable(conf);
> >                          isReadOnly = isReadOnly || !symbols.getFile().canWrite();
> >
> > +                        try {
> > +                            // initialize EXPath repository so indexManager and
> > +                            // startup triggers can access it
> > +                            expathRepo = ExistRepository.getRepository(this.conf);
> > +                        } catch(final PackageException e) {
> > +                            LOG.warn("Failed to initialize expath repository: " + e.getMessage() + " - " +
>
> +1 to throw exception, but it also require a way to repair such exception
> ... thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eXist-db/exist/pull/726/files#r34247002>.
>



-- 
Adam Retter

skype: adam.retter
tweet: adamretter
http://www.adamretter.org.uk